File: ps/minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minimum=1000000000;
    int minDepth(TreeNode* root) {
        if (root == NULL) return 0;
        recurse(root, 1);
        return minimum;
    }
    void recurse(TreeNode *cur, int depth){
        if (cur == NULL) return;
        if (cur->right == NULL && cur->left == NULL) {
            minimum = min(depth, minimum);
            return;
        }
        recurse(cur->left, depth+1);
        recurse(cur->right, depth+1);
        return;
    }
};

File: ps/minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        if root.left is None and root.right:
            return self.minDepth(root.right) + 1
        if root.right is None and root.left:
            return self.minDepth(root.left) + 1

        return min(self.minDepth(root.left) + 1, self.minDepth(root.right) + 1)
